README.md

```diff
@@ -37,124 +71,378 @@ To install the global command line interface (CLI), use npm global installation:
 npm install -g fqm-execution
 ```
 
-## Usage
+# Usage
+
+## Quickstart
+
+```typescript
+import { Calculator } from 'fqm-execution';
+
+// ...
 
-### API
+const { results } = await Calculator.calculate(measureBundle, patientBundles, options[, valueSetCache]);
 
-Import the necessary modules:
+// ...
+```
+
+## Requirements
+
+To calculate a FHIR-based eCQM, `fqm-execution` needs the following information:
+
+- A [FHIR Bundle](http://hl7.org/fhir/bundle.html) resource that contains:
+  - One [FHIR Measure](https://build.fhir.org/ig/HL7/cqf-measures/StructureDefinition-measure-cqfm.html) resource containing the metadata for the measure
+  - The [FHIR Library](https://build.fhir.org/ig/HL7/cqf-measures/StructureDefinition-library-cqfm.html) resource that is referenced from the above `Measure` resource's `.library` property, and any other FHIR `Library` resources that are dependencies of the main measure logic (e.g. `FHIRHelpers`)
+    - **NOTE**: These `Library` resources _must_ contain `base64`-encoded ELM JSON content directly on the resource. `fqm-execution` does not do any real-time translation of CQL to ELM
+  - Any [FHIR ValueSet](http://hl7.org/fhir/valueset.html) resources that are used in the measure logic\*
+- One or more FHIR `Bundle` resources that contain:
+  - One [FHIR Patient](http://hl7.org/fhir/patient.html) resource
+  - Any other FHIR resources that contain relevant data for the above patient that should be considered during measure calculation
+
+\*`ValueSet` resources can be omitted if you follow the approach outlined in the [ValueSet Resolution](#valueset-resolution) section
+
+## Calculating an eCQM
+
+`fqm-execution` exports the `Calculator` object, which contains an API function that returns eCQM calculation results as a detailed object containing calculation results for each patient across all population groups defined in `Measure` resource:
+
+```typescript
+import { Calculator } from 'fqm-execution';
+
+// ...
+
+const { results } = await Calculator.calculate(measureBundle, patientBundles, options[, valueSetCache]);
+/*        ^?
```

Comment:
For this first one, yeah it makes sense to separate since it's the whole structure. For the smaller examples later on, I think it's fine to leave as is since people who are in those sections likely have more exposure to results already